source/docs/software/commandbased/commands-v3/what-is-command-based-v3.rst

```diff
@@ -0,0 +1,241 @@
+
+.. code-block:: java
+
+   Command extendWhenReady = Command.noRequirements()
+     .executing(coroutine -> {
+       while (!condition.get()) {
+         coroutine.yield(); // Wait for condition
+       }
+       piston.set(Value.kForward);
+     })
+     .named("Extend When Ready");
+
+**Without command-based (manual state tracking):**
+
+.. code-block:: java
+
+   // In periodic():
+   if (condition.get()) {
+     if (!pressed) {
+       piston.set(Value.kForward);
+       pressed = true;
+     }
+   } else {
+     pressed = false;
+   }
+
+The declarative v3 trigger style is concise for simple actions. The imperative v3 coroutine style shines when you have complex sequential logic, loops, or conditionals.
+
+## Mechanisms and Commands
+
+.. image:: ../diagrams/subsystems-and-commands.drawio.svg
+   :alt: Mechanisms and commands diagram
+
+Commands v3 is built around two core abstractions: **commands** and **mechanisms**.
+
+### Commands
+
+**Commands** represent actions the robot can take. In v3, you can write commands either:
+
+1. **Declaratively** (like v2): Compose commands using ``.sequence()``, ``.alongWith()``, etc.
+2. **Imperatively** (new in v3): Write sequential code with loops, conditionals, and ``coroutine.yield()``
+
+Commands run when scheduled, until they are interrupted or their end condition is met. Commands are recursively composable: you can combine simple commands to create complex behaviors.
+
+### Mechanisms
+
+**Mechanisms** (called "Subsystems" in v2) represent independently-controlled collections of robot hardware. Mechanisms serve two purposes:
+
+1. **Encapsulation**: Hide hardware implementation details from the rest of your code
+2. **Resource management**: Only one command can use a mechanism at a time, preventing conflicts
+
+For example, a ``Drivetrain`` mechanism might contain motor controllers and encoders. Commands that require the ``Drivetrain`` automatically coordinate through the scheduler to avoid conflicts.
+
+See :ref:`docs/software/commandbased/commands-v3/mechanisms:Mechanisms` for detailed documentation.
+
+## How Commands Are Run
+
+Commands are run by the ``Scheduler`` singleton, which:
+
+- Polls triggers (buttons, conditions) to schedule commands
+- Prevents resource conflicts (only one command per mechanism)
+- Executes scheduled commands cooperatively using coroutines
+
+Call ``Scheduler.getDefault().run()`` from ``robotPeriodic()`` in your ``Robot`` class. This runs at 50Hz (every 20ms) by default.
+
+.. code-block:: java
+
+   import org.wpilib.commands3.Scheduler;
+   import edu.wpi.first.wpilibj.TimedRobot;
+
+   public class Robot extends TimedRobot {
+     @Override
+     public void robotPeriodic() {
+       Scheduler.getDefault().run();
+     }
+   }
+
+Multiple commands can run concurrently, as long as they don't require the same mechanisms. The scheduler uses priority levels to decide which command runs when there's a conflict.
+
+## What's Different in v3?
```

Comment:
Let's structure this as "what are the important concepts" instead of "what's different"; that's better done in a migration guide than the introduction to the framework